src/components/Charts/Charts.tsx

```diff
@@ -67,47 +79,67 @@ export default function Charts({
 			}
 		}
 
-		setChartData(newChartData);
 		console.log("Chart data: ", newChartData);
+		setChartData(newChartData);
 	}, [cleanTransactions]);
 
-	const renderChart = (
-		<ResponsiveContainer width="90%" height={500}>
-			<LineChart
-				margin={{ top: 10, right: 10, bottom: 10, left: 10 }}
-				data={chartData}
-			>
-				<CartesianGrid stroke="#C1D2D9" strokeDasharray="5 5" />
-				<XAxis dataKey="month" />
-				<YAxis />
-				<Legend />
-				<Tooltip />
-				{configGroups.map((group) => {
-					if (chartData[0] === undefined) return <></>;
-					return (
-						<Line
-							type="monotone"
-							dataKey={group.name}
-							stroke="#AC3931"
-							dot={{ stroke: "#AC3931", strokeWidth: 1 }}
-						></Line>
-					);
-				})}
-				<Line
-					type="monotone"
-					dataKey="Income"
-					stroke="#537D8D"
-					dot={{ stroke: "#537D8D", strokeWidth: 1 }}
-				></Line>
-			</LineChart>
-		</ResponsiveContainer>
-	);
+	const checkboxTicked = (event: React.ChangeEvent<HTMLInputElement>) => {
+		setDataToDisplay({
+			...dataToDisplay,
+			[event.target.name]: event.target.checked,
+		});
+	};
 
 	// RENDER
 	return (
 		<Container>
 			<h2>Chart</h2>
-			{renderChart}
+			<Grid container spacing={3}>
+				<Grid item xs={10}>
+					<ResponsiveContainer width="95%" height={500}>
+						<LineChart
+							margin={{ top: 10, right: 10, bottom: 10, left: 10 }}
+							data={chartData}
+						>
+							<CartesianGrid stroke="#C1D2D9" strokeDasharray="5 5" />
+							<XAxis dataKey="month" />
+							<YAxis />
+							<Tooltip />
+							{Object.keys(dataToDisplay).reduce(
```

Comment:
C'est pas très lisible ceci en ligne.

Si c'est pas possible d'avoir les lines en dehors, peut-être crée une structure de donnée et applique là ensuite dans le jsx.

Egalement, reduce rend cette partie super obscure...

Reply:
Replaced the reduce by filter + map for added clarity.